config/setup/errors.go

```diff
@@ -21,12 +23,17 @@ func Errors(c *Controller) (middleware.Middleware, error) {
 	// Open the log file for writing when the server starts
 	c.Startup = append(c.Startup, func() error {
 		var err error
-		var file *os.File
+		var file io.Writer
 
 		if handler.LogFile == "stdout" {
 			file = os.Stdout
 		} else if handler.LogFile == "stderr" {
 			file = os.Stderr
+		} else if handler.LogFile == "syslog" {
+			file, err = gsyslog.NewLogger(gsyslog.LOG_ERR, "SYSLOG", "caddy")
```

Comment:
anyway when parsing syslog lines, one usually does not rely on the
facility, but rather on the "syslog-tag" (called "app-name" in recent
version of syslog protocol). the syslog tag should just be "caddy[pid]".
local0 does seem fine for facility.

regards,
stephane
Le 31 juil. 2015 15:10, "Josh Wood" <notifications@github.com> a écrit :

> In config/setup/errors.go
> <https://github.com/mholt/caddy/pull/210#discussion_r35971364>:
>
> >
> >  		if handler.LogFile == "stdout" {
> >  			file = os.Stdout
> >  		} else if handler.LogFile == "stderr" {
> >  			file = os.Stderr
> > +		} else if handler.LogFile == "syslog" {
> > +			file, err = gsyslog.NewLogger(gsyslog.LOG_ERR, "SYSLOG", "caddy")
>
> From caddy's point of view, the *n* in LOCALn is arbitrary. LOCAL0 works
> for me.
>
> All *facility* gives you is a string to match in syslog.conf -- to treat
> lines from different facilities in different ways. So caddy's syslog
> facility need only be known and documented. Despite suggesting it above, I
> don't really think log facility needs to be configurable, at least not now
> in this PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mholt/caddy/pull/210/files#r35971364>.
>